tracker/src/main/java/org/matomo/sdk/dispatcher/DefaultDispatcher.java

```diff
@@ -139,6 +151,44 @@ public boolean forceDispatch() {
         return true;
     }
 
+    @Override
+    public void forceDispatchBlocking() {
+        Thread dispatchThread = null;
+
+        synchronized (mThreadControl) {
+            // ensure the dispatch loop always exits after a single loop
+            mForcedBlocking = true;
+
+            if (mRunning) {
```

Comment:
I started down that route but ended going this way for two reasons.  The first was that I really didn't want to spin up a new thread during the UncaughtException path.  The second is that `forceDispatch` either trips the semaphore or launches the dispatch thread which will block until the dispatch interval hits.  But it doesn't do both.  I needed to make sure the semaphore was released and we don't end up waiting on the interval.  I also tried to go down the route of having the dispatch thread ensure that everything is flushed so you don't have to double invoke the loop and it made things kinda messy, so I backed off and wrote the blocking path to be independent.

If the duplication is too much I can take another pass but I'd like to understand more about the reasoning behind how `launch()` was written before I do so I don't break existing function.

Reply:
I took another look at this and called `forceDispatch` .  I was trying to handle the case where more events came in after the crash initially.  On second look we were just passing the unit test I created for it because a second call was happening after the event generator ended. (Not a valid test).

I updated the code in the pr update.